include/h2o/balancer.h

```diff
@@ -41,16 +41,38 @@ typedef struct st_h2o_balancer_callbacks_t {
     h2o_balancer_destroyer destroy;
 } h2o_balancer_callbacks_t;
 
+typedef enum en_h2o_balancer_type_t {
+    H2O_BALANCER_TYPE_ROUND_ROBIN,
+    H2O_BALANCER_TYPE_LEAST_CONN,
+    H2O_BALANCER_TYPE_HASH
+} h2o_balancer_type_t;
+
 struct st_h2o_balancer_t {
     const h2o_balancer_callbacks_t *callbacks;
+    const int idempotent;
```

Comment:
Would you mind elaborating what the meaning of the variable is and why it needs to be a common variable shared to all the load balancing algorithms?

It is a bit surprising to see such a variable being added in a PR that adds a new algorithm.